### ISSUE TITLE
Linuxbrew's superenv will pass unwanted flags to compiler

### ISSUE THREAD

**Issue by xu-cheng** [Homebrew/homebrew-portable#32]
As Linuxbrew recently enables superenv, it will unconditionally pass some unwanted flags to the compiler. These flags are list below, which are important to normal formulae but not formulae in this repo. 
```
-Wl,--dynamic-linker=#{HOMEBREW_PREFIX}/lib/ld.so
-Wl,-rpath,#{HOMEBREW_PREFIX}/lib
```
Noted that we always want portable formulae being linked to system glibc provided by CentOS instead of the one provided by Linuxbrew for compatible reason.

Previously, these flags are reset by our formulae when Linuxbrew uses stdenv.

Currently, since `#{HOMEBREW_PREFIX}/lib/ld.so` does not exist in the docker container. Nothing is broken. However, it would still be nice to prevent passing them in some way. For example, a flag like `HOMEBREW_NO_LINUXBREW_GLIBC`.

cc @sjackman.

**Comment by sjackman** [comment]
When building `glibc` it needs a similar fix that doesn't yet exist, to prevent it from using `#{HOMEBREW_PREFIX}/lib/ld.so` which is a symlink to the host's dynamic linker before `glibc` has finished being built.

The quick fix is to add `env :std if OS.linux?` to your portable formulae.

Superenv passes the `RPATH` paths in `HOMEBREW_RPATH_PATHS`. We could pass the desired dynamic linker in a similar environment variable `HOMEBREW_DYNAMIC_LINKER`.

You say the `ld.so` symlink isn't breaking anything for you. Is the `RPATH` issue causing breakage from you? You should be able to simply remove that path from `HOMEBREW_RPATH_PATHS`. I haven't tried it, but seems to me it should work.

**Comment by xu-cheng** [comment]
>  Is the RPATH issue causing breakage from you?

Nothing is broken, which to be frankly surprises me a bit.

FYI, The only tweak I added is to set `HOMEBREW_CC` to `cc`.

> The quick fix is to add env :std if OS.linux? to your portable formulae.

In fact, I'd like to make them compiled in suprenv, which may provide its own benefits. And since nothing is broken, I don't think we need this for now.

**Comment by xu-cheng** [comment]
> We could pass the desired dynamic linker in a similar environment variable HOMEBREW_DYNAMIC_LINKER.

This sounds good to me.

**Comment by sjackman** [comment]
>> Is the RPATH issue causing breakage from you?

> Nothing is broken, which to be frankly surprises me a bit.

I believe you're linking all the dependent libraries, except `libc`, which you're linking dynamically? Does the system that you're testing on have the brewed `glibc` formula installed?

**Comment by sjackman** [comment]
>> We could pass the desired dynamic linker in a similar environment variable HOMEBREW_DYNAMIC_LINKER.

> This sounds good to me.

It won't be a priority for me if it's not currently broken for you, but it's probably the solution I'll use when I fix `glibc` for `--env=super`.

**Comment by xu-cheng** [comment]
@sjackman https://github.com/Linuxbrew/brew/pull/347

**Comment by sjackman** [comment]
Fixed by Linuxbrew/brew#347 when it's merged.
